yt_dlp/extractor/tiktok.py

```diff
@@ -51,7 +51,17 @@ class TikTokBaseIE(InfoExtractor):
 
     @property
     def _KNOWN_APP_INFO(self):
-        return self._configuration_arg('app_info', ie_key=TikTokIE)
+        # If we have a genuine device ID, we may not need any IID
+        default = [''] if self._KNOWN_DEVICE_ID else []
+        return self._configuration_arg('app_info', default, ie_key=TikTokIE)
+
+    @property
+    def _KNOWN_DEVICE_ID(self):
+        return self._configuration_arg('device_id', [None], ie_key=TikTokIE)[0]
+
+    @property
+    def _DEVICE_ID(self):
+        return self._KNOWN_DEVICE_ID or str(random.randint(7250000000000000000, 7351147085025500000))
```

Comment:
using lower case makes more semantic sense, no?

Reply:
with 04574eb7a619e4a1fe5519b0ae77755cd1960c89 I think uppercase makes sense, since they are effectively class constants then (and we maintain consistency with `_API_HOSTNAME` etc)